syft/file/cataloger/executable/cataloger.go

```diff
@@ -165,16 +167,19 @@ func processExecutable(loc file.Location, reader unionreader.UnionReader) (*file
 
 	switch format {
 	case file.ELF:
-		if err := findELFFeatures(&data, reader); err != nil {
+		if err = findELFFeatures(&data, reader); err != nil {
 			log.WithFields("error", err).Tracef("unable to determine ELF features for %q", loc.RealPath)
+			err = fmt.Errorf("unable to determine ELF features: %w", err)
```

Comment:
```suggestion
			err = fmt.Errorf("unable to determine ELF features for %q: %w", loc.Path(), err)
```

same comment applies to surrounding cases too

Reply:
nvm -- this is already part of the file object *facepalm*